### PR TITLE
Dev container.

### PR DESCRIPTION
**Pull request by artursouza**
Dev container for predictable setup.

### How has this code been tested?
Opened in VS Code, then: make -j 4

### Checklist

I have:
- [X] Run `make reviewable` to ensure this PR is ready for review.
- [X] Ensured this PR contains a neat, self documenting set of commits.
- [X] Updated any relevant [documentation] and [examples].
- [ ] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

### CODE REVIEW

**Comment by negz** [comment]
Thanks for suggesting this @artursouza! Could you speak a little as to what problems you encountered that lead you to develop this? I believe our Make library is already intended to be at least partly independent of the system configuration (i.e. it pins and caches specific versions of most tools), but I understand that we're not currently completely hermetic. One example of this is that we'll try to use the system Go installation.

I'm going to assign this to @jbw976 and @bassam, who have the deepest understanding of and passion for our Makefile library and may have opinions about using said library from within a Docker image.

File: go.mod [hunk 1]
```diff
@@ -3,21 +3,54 @@ module github.com/crossplane/crossplane
...
+	github.com/yvasiyarov/gorelic v0.0.7 // indirect
+	github.com/yvasiyarov/newrelic_platform_go v0.0.0-20160601141957-9c099fbc30e9 // indirect
+	go.pedge.io/lion v0.0.0-20190619200210-304b2f426641 // indirect
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
+	gopkg.in/inconshreveable/log15.v2 v2.0.0-20200109203555-b30bc20e4fd1 // indirect
 	k8s.io/api v0.17.3
 	k8s.io/apiextensions-apiserver v0.0.0-20190918161926-8f644eb6e783
 	k8s.io/apimachinery v0.17.3
 	k8s.io/client-go v0.17.3
+	rsc.io/letsencrypt v0.0.3 // indirect
```
Comment (negz) [inline go.mod]: It seems this is currently failing due to `make check-diff`; likely because `go mod tidy` would remove some of these indirect dependencies.

Comment (artursouza) [reply go.mod]: I am fixing this now.

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,109 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
```
Comment (negz) [inline .devcontainer/Dockerfile]: Also, just caught this. Can we MIT license this and copyright it to Microsoft under the current Crossplane license and governance? I suspect we will at least have to use the Apache 2.0 license the rest of the project uses, and attribute copyright to the Crossplane authors.

Comment (artursouza) [reply .devcontainer/Dockerfile]: Good point. Let me adhere to the existing license.

**Comment by artursouza** [comment]
Without dev container, build fails consistently on my macOS Catalina with Golang 1.14. To avoid resetting my dev environment for crossplane only (since I have other project to support in Go v1.14), I preferred to use dev container so all the requirements to code for Crossplane are self contained.
```
[15:52:32] ~/Code/crossplane/crossplane (dev_container)
$ export GO_SUPPORTED_VERSIONS=1.14
[15:53:08] ~/Code/crossplane/crossplane (dev_container)
$ make -j 4
15:53:10 [ .. ] verify dependencies have expected content
==> Linting /Users/asouza/Code/crossplane/crossplane/cluster/charts/crossplane-controllers
Lint OK

1 chart(s) linted, no failures
==> Linting /Users/asouza/Code/crossplane/crossplane/cluster/charts/crossplane-types
Lint OK

1 chart(s) linted, no failures
==> Linting /Users/asouza/Code/crossplane/crossplane/cluster/charts/crossplane
Lint OK

1 chart(s) linted, no failures
all modules verified
15:53:21 [ OK ] go modules dependencies verified
15:53:21 [ .. ] helm dep crossplane 0.8.0-rc.94.g2359b574-dirty
15:53:21 [ .. ] helm dep crossplane-types 0.8.0-rc.94.g2359b574-dirty
15:53:21 [ .. ] helm dep crossplane-controllers 0.8.0-rc.94.g2359b574-dirty
15:53:21 [ .. ] golangci-lint
No requirements found in /Users/asouza/Code/crossplane/crossplane/cluster/charts/crossplane-controllers/charts.
15:53:21 [ OK ] helm dep crossplane-controllers 0.8.0-rc.94.g2359b574-dirty
No requirements found in /Users/asouza/Code/crossplane/crossplane/cluster/charts/crossplane-types/charts.
No requirements found in /Users/asouza/Code/crossplane/crossplane/cluster/charts/crossplane/charts.
15:53:21 [ OK ] helm dep crossplane-types 0.8.0-rc.94.g2359b574-dirty
15:53:21 [ OK ] helm dep crossplane 0.8.0-rc.94.g2359b574-dirty
WARN [runner] Can't run linter goanalysis_metalinter: unparam: failed prerequisites: buildssa@github.com/crossplane/crossplane/apis/cache/v1alpha1
WARN [runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/rediscluster_types.go:20:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool))))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/rediscluster_types.go:22:18: could not import github.com/crossplane/crossplane-runtime/apis/core/v1alpha1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/github.com/crossplane/crossplane-runtime@v0.5.1-0.20200303183616-3d4c771a3062/apis/core/v1alpha1/condition.go:20:2: could not import sort (/usr/local/Cellar/go/1.14/libexec/src/sort/slice_go113.go:9:8: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/register.go:20:2: could not import reflect (/usr/local/Cellar/go/1.14/libexec/src/reflect/type.go:19:2: could not import strconv (/usr/local/Cellar/go/1.14/libexec/src/strconv/atof.go:13:8: could not import math (/usr/local/Cellar/go/1.14/libexec/src/math/exp_asm.go:9:8: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/register.go:22:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/register.go:23:2: could not import sigs.k8s.io/controller-runtime/pkg/scheme (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/scheme/scheme.go:58:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:23:9: could not import k8s.io/api/core/v1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/api@v0.17.3/core/v1/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.deepcopy.go:24:10: could not import k8s.io/apimachinery/pkg/runtime (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/codec.go:20:2: could not import bytes (/usr/local/Cellar/go/1.14/libexec/src/bytes/buffer.go:10:2: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:84:10: cm.Spec.WriteConnectionSecretToReference undefined (type RedisClusterSpec has no field or method WriteConnectionSecretToReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:79:10: cm.Spec.ResourceReference undefined (type RedisClusterSpec has no field or method ResourceReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:69:10: cm.Spec.ClassSelector undefined (type RedisClusterSpec has no field or method ClassSelector) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:64:10: cm.Spec.ClassReference undefined (type RedisClusterSpec has no field or method ClassReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:54:17: cm.Spec.WriteConnectionSecretToReference undefined (type RedisClusterSpec has no field or method WriteConnectionSecretToReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:49:17: cm.Spec.ResourceReference undefined (type RedisClusterSpec has no field or method ResourceReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:39:17: cm.Spec.ClassSelector undefined (type RedisClusterSpec has no field or method ClassSelector) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:34:17: cm.Spec.ClassReference undefined (type RedisClusterSpec has no field or method ClassReference)]
ERRO Running error: buildssa: analysis skipped: errors in package: [/Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/rediscluster_types.go:20:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool))))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/rediscluster_types.go:22:18: could not import github.com/crossplane/crossplane-runtime/apis/core/v1alpha1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/github.com/crossplane/crossplane-runtime@v0.5.1-0.20200303183616-3d4c771a3062/apis/core/v1alpha1/condition.go:20:2: could not import sort (/usr/local/Cellar/go/1.14/libexec/src/sort/slice_go113.go:9:8: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/register.go:20:2: could not import reflect (/usr/local/Cellar/go/1.14/libexec/src/reflect/type.go:19:2: could not import strconv (/usr/local/Cellar/go/1.14/libexec/src/strconv/atof.go:13:8: could not import math (/usr/local/Cellar/go/1.14/libexec/src/math/exp_asm.go:9:8: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/register.go:22:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/register.go:23:2: could not import sigs.k8s.io/controller-runtime/pkg/scheme (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/scheme/scheme.go:58:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:23:9: could not import k8s.io/api/core/v1 (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/api@v0.17.3/core/v1/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.14/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.deepcopy.go:24:10: could not import k8s.io/apimachinery/pkg/runtime (/Users/asouza/Code/crossplane/crossplane/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/runtime/codec.go:20:2: could not import bytes (/usr/local/Cellar/go/1.14/libexec/src/bytes/buffer.go:10:2: could not import errors (/usr/local/Cellar/go/1.14/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.14/libexec/src/internal/reflectlite/value.go:8:2: could not import runtime (/usr/local/Cellar/go/1.14/libexec/src/runtime/alg.go:8:2: could not import internal/cpu (-: could not load export data: cannot import "internal/cpu" (unknown iexport format version 1), export data is newer version - update tool)))))) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:84:10: cm.Spec.WriteConnectionSecretToReference undefined (type RedisClusterSpec has no field or method WriteConnectionSecretToReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:79:10: cm.Spec.ResourceReference undefined (type RedisClusterSpec has no field or method ResourceReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:69:10: cm.Spec.ClassSelector undefined (type RedisClusterSpec has no field or method ClassSelector) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:64:10: cm.Spec.ClassReference undefined (type RedisClusterSpec has no field or method ClassReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:54:17: cm.Spec.WriteConnectionSecretToReference undefined (type RedisClusterSpec has no field or method WriteConnectionSecretToReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:49:17: cm.Spec.ResourceReference undefined (type RedisClusterSpec has no field or method ResourceReference) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:39:17: cm.Spec.ClassSelector undefined (type RedisClusterSpec has no field or method ClassSelector) /Users/asouza/Code/crossplane/crossplane/apis/cache/v1alpha1/zz_generated.claim.go:34:17: cm.Spec.ClassReference undefined (type RedisClusterSpec has no field or method ClassReference)]
15:53:35 [FAIL]
make[2]: *** [go.lint] Error 1
make[1]: *** [build.all] Error 2
make: *** [build] Error 2
```

**Comment by artursouza** [comment]
I found the fix to allow me to build on my dev machine with go 1.14: https://github.com/upbound/build/pull/94

We don't need this dev machine necessarily. It is a nice to have.

File: go.mod [hunk 3]
```diff
@@ -3,16 +3,20 @@ module github.com/crossplane/crossplane
 go 1.13
 
 require (
+	cloud.google.com/go/pubsub v1.3.1 // indirect
```
Comment (jbw976) [inline go.mod]: any idea why this change would be bringing in new dependencies to our build? I don't see any `*.go` files changing, so I'm not sure why we're getting new dependencies 🤔

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    docker.io \
+    jq \
+    make \
+    rsync \
+    runit \
+    unzip \
+    zip \
+    && mkdir -p /tmp/gotools \
+    && cd /tmp/gotools \
+    && GO111MODULE=on go get -v golang.org/x/tools/gopls@latest 2>&1 \
```
Comment (jbw976) [inline .devcontainer/Dockerfile]: what's your thoughts on using `latest` tags for all of these imports within the dev container? That means that depending on when a community members ends up using this for the first time, they will get a different set of versions for the tools, which makes this a less consistent and repeatable experience.  Do you think that will be an issue?

File: .devcontainer/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ====================================================================================
+
+FROM golang:1.13-buster
```
Comment (jbw976) [inline .devcontainer/Dockerfile]: what's the maintenance burden of this file over time? Will we need to check in on it every so often, or wait till someone says it's not working for them, or how do you see that going?

File: go.mod (hunk 3 above)
Comment (artursouza) [reply go.mod]: I can remove that. It might be after doing go mod tidy in an older version.

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (artursouza) [reply .devcontainer/Dockerfile]: I agree. We should stick to a version tag whenever is possible.

File: .devcontainer/Dockerfile (hunk 5 above)
Comment (artursouza) [reply .devcontainer/Dockerfile]: I don’t really have enough time using dev containers to answer that. I do believe we can add it to the CICD pipeline and automate its validation via GitHub actions.

On the other hand, one thing that is holding me back to proceed with this change is that each repo needs to have its own dev container. It can be a Git module but then it becomes too much overhead IMO.

The upgrade to Go 1.14 is the right thing rondo if we can support 1.13 as well.

**Comment by artursouza** [comment]
I am closing this PR. It can serve as a reference if we want to bring the dev container idea back.
